### PR TITLE
[RFC] legacy tests: migrate test38

### PR DESCRIPTION
**Pull request by muralisc**
Migrated test38 to test/functional/legacy/038_virtual_replace_spec.lua ( Issue #1286 )
This is my first contribution to an open source project, let me know if I did something wrong !!

### CODE REVIEW

File: test/functional/legacy/038_virtual_replace_spec.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('Virtual replace mode', function()
+  setup(clear)
+
+  it('is working', function()
+    -- Make sure that backspace works, no matter what termcap is used.
+    execute('set t_kD=x7f t_kb=x08')
```
Comment (glts) [inline test/functional/legacy/038_virtual_replace_spec.lua]: This line contains two actual control characters `^V`. Our helpers support angle bracket notation, so you can replace this line with 

    execute('set t_kD=<C-v>x7f t_kb=<C-v>x08')

File: test/functional/legacy/038_virtual_replace_spec.lua [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    feed('ggdGa<cr>')
+    feed('abcdefghi<cr>')
+    feed('jk<tab>lmn<cr>')
+    feed('<Space><Space><Space><Space>opq<tab>rst<cr>')
+    feed('<C-d>uvwxyz<cr>')
+    feed('<esc>gg:set ai<cr>')
```
Comment (glts) [inline test/functional/legacy/038_virtual_replace_spec.lua]: Sometimes a little manual clean-up makes the test easier to read afterwards. You could align the `:set` here with the one in the next line by splitting this line into two:

    feed('<esc>gg')
    execute('set ai')

File: test/functional/legacy/038_virtual_replace_spec.lua [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    feed('A<cr>')
+    feed('BCDEFGHIJ<cr>')
+    feed('<tab>KL<cr>')
+    feed('MNO<cr>')
+    feed('PQR<C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><C-h><esc>:$<cr>')
+    feed('iab<tab>cdefghi<tab>jkl<esc>0gRAB......CDEFGHI.J<esc>o<esc>:<cr>')
+    feed('iabcdefghijklmnopqrst<esc>0gRAB<tab>IJKLMNO<tab>QR<esc>')
+
+    -- Assert buffer contents.
+    expect([=[
```
Comment (glts) [inline test/functional/legacy/038_virtual_replace_spec.lua]: You can indent this just like the rest, `expect` will strip the leading whitespace.

    expect([[
       1
       A
       BCDEFGHIJ
       	KL
      	MNO
      	PQR
       1
      abcdefghi
      ...

**Comment by glts** [comment]
Thanks a lot, @muralisc. I left some comments, otherwise this looks good to me.

File: test/functional/legacy/038_virtual_replace_spec.lua (hunk 1 above)
Comment (glts) [inline test/functional/legacy/038_virtual_replace_spec.lua]: Actually I don't think `execute` supports this, it's probably fine as it is :)

File: test/functional/legacy/038_virtual_replace_spec.lua (hunk 3 above)
Comment (muralisc) [inline test/functional/legacy/038_virtual_replace_spec.lua]: restored the indent !!

File: test/functional/legacy/038_virtual_replace_spec.lua (hunk 2 above)
Comment (muralisc) [inline test/functional/legacy/038_virtual_replace_spec.lua]: done !

**Comment by justinmk** [comment]
Merged, thanks!

@muralisc , if you plan to make changes to the core source code, please sign the CLA.

**Comment by muralisc** [comment]
Signed the CLA , Thanks guys .

**Pull request closed without merging**